pkg/operator/ceph/cluster/mon/config.go

```diff
@@ -319,3 +322,42 @@ func ExtractKey(contents string) (string, error) {
 	}
 	return secret, nil
 }
+
+// PopulateExternalClusterInfo Add validation in the code to fail if the external cluster has no OSDs keep waiting
+func PopulateExternalClusterInfo(context *clusterd.Context, namespace string) *cephconfig.ClusterInfo {
+	var clusterInfo *cephconfig.ClusterInfo
+	for {
+		var err error
+		clusterInfo, _, _, err = LoadClusterInfo(context, namespace)
+		if err != nil {
+			logger.Warningf("waiting for the connection info of the external cluster. retrying in %s.", externalConnectionRetry.String())
+			time.Sleep(externalConnectionRetry)
```

Comment:
Instead of the `for` loop and sleeping forever, shouldn't we be re-queueing the event?

Reply:
I'm just re-using the old code, I cannot quite do this change since it will affect the CephCluster CR too, and in 1.3 it's not controller runtime yet.